src/golang/lib/workflow/dag/response.go

```diff
@@ -0,0 +1,123 @@
+package dag
+
+import (
+	"time"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/artifact_result"
+	"github.com/aqueducthq/aqueduct/lib/collections/operator_result"
+	"github.com/aqueducthq/aqueduct/lib/collections/shared"
+	"github.com/aqueducthq/aqueduct/lib/collections/workflow"
+	"github.com/aqueducthq/aqueduct/lib/collections/workflow_dag"
+	"github.com/aqueducthq/aqueduct/lib/collections/workflow_dag_result"
+	"github.com/aqueducthq/aqueduct/lib/workflow/artifact"
+	"github.com/aqueducthq/aqueduct/lib/workflow/operator"
+	"github.com/google/uuid"
+)
+
+type MetadataResponse struct {
+	// Dag metadata
+	DagId         uuid.UUID             `json:"dag_id"`
+	DagCreatedAt  time.Time             `json:"dag_created_at"`
+	StorageConfig *shared.StorageConfig `json:"storage_config"`
+	EngineConfig  *shared.EngineConfig  `json:"engine_config"`
+
+	// Workflow metadata
+	WorkflowId        uuid.UUID                 `json:"workflow_id"`
+	WorkflowCreatedAt time.Time                 `json:"workflow_created_at"`
+	UserId            uuid.UUID                 `json:"user_id"`
+	Name              string                    `json:"name"`
+	Description       string                    `json:"description"`
+	Schedule          *workflow.Schedule        `json:"schedule"`
+	RetentionPolicy   *workflow.RetentionPolicy `json:"retention_policy"`
+}
+
+type Response struct {
```

Comment:
Where is this used?

Reply:
It's just for consistency so that each wf / op / artf object has one response for 'metadata' and one response for 'results'. We will need it in future for sure so I think there's no harm defining it now.